src/service/matchers/lineage_matcher.py

```diff
@@ -3,39 +3,46 @@
 """
 
 import logging
+import re
+from typing import Any, Self
 
-from pydantic import ConfigDict, BaseModel, Field
-from typing import Any
+from pydantic import ConfigDict, BaseModel, Field, model_validator
 
 from src.common.constants import GTDB_UNCLASSIFIED_PREFIX
 from src.common.gtdb_lineage import GTDBRank, GTDBLineage, GTDBLineageError
 from src.common.storage.collection_and_field_names import FLD_GENOME_ATTRIBS_GTDB_LINEAGE
 from src.service import errors
 from src.service import models
 from src.service.app_state_data_structures import PickleableDependencies
-from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 from src.service.data_products import genome_attributes
 from src.service.matchers.common_models import Matcher
-
+from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 
 # See the workspace specs for the types listed in the matcher
 _GTDB_LINEAGE_METADATA_KEY = "GTDB_lineage"
 _GTDB_LINEAGE_VERSION_METADATA_KEY = "GTDB_source_ver"
 
-_COLLECTION_PARAMS_GTDB_VERSION_KEY = "gtdb_version"
+_COLLECTION_PARAMS_GTDB_VERSIONS_KEY = "gtdb_versions"
 
 
 class GTDBLineageMatcherCollectionParameters(BaseModel):
     "Parameters for the GTDB lineage matcher."
-    gtdb_version: str = Field(
-        example="207.0",
-        description="The GTDB version of the collection in which the matcher is installed. " +
-            "Input data to the matcher must match this version of GTDB or the match will " +
+    gtdb_versions: list[str] = Field(
+        example=["214.0", "214.1"],
+        description="The allowed GTDB versions of the collection in which the matcher is installed. " +
+            "Input data to the matcher must match one of the allowed versions of GTDB or the match will " +
             "abort.",
-        pattern=r"^\d{2,4}\.\d{1,2}$"  # giving a little room for expansion
```

Comment:
https://stackoverflow.com/questions/58098925/use-regex-in-pydantic-model-that-works-with-set-of-strings

Reply:
👍